01 react concepts/6/03 lifecycle

crsanti opened the pull request:
This closes #6.

crsanti committed:
Initial configuration.

crsanti committed:
Added ColorPicker

crsanti committed:
Added bootstrap colorpicker

crsanti committed:
Added bootstrap-colorpicker.
Removed bootstrap-colorpicker-plus

crsanti committed:
Removed unnecesary state and methods.

crsanti committed:
Added README.md doc

Removed link in 02 State

crsanti committed:
Removed 'bootstrap-colorpicker' from vendor bundle.

crsanti committed:
Removed unnecessary alias

crsanti committed:
Merge branch 'master' into 01_React_Concepts/6/03_Lifecycle

nasdan commented on 01 React Concepts/03 LifeCycle/src/components/App.tsx:
```diff
@@ -0,0 +1,107 @@
+  Moon: '#466579',
+};
+
+
```
remove space

nasdan commented on 01 React Concepts/03 LifeCycle/src/components/App.tsx:
```diff
@@ -0,0 +1,107 @@
+        <header>
+          <h2 className="col-sm-6 col-sm-offset-1">React component lifecycle methods</h2>
+        </header>
+        <form className="form-horizontal" onSubmit={this.sendForm}>
```
Extract into subcomponent

nasdan commented on 01 React Concepts/03 LifeCycle/src/components/App.tsx:
```diff
@@ -0,0 +1,107 @@
+                    onClick={this.onCommonColorPick(commonColors.Darkgreen)}
+                    className="btn btn-default white"
+                    style={{ backgroundColor: commonColors.Darkgreen }}>Darkgreen</button>
+                  <button
```
Extract into subcomponent

nasdan commented on 01 React Concepts/03 LifeCycle/src/components/ColorPicker.tsx:
```diff
@@ -0,0 +1,72 @@
+  }
+
+  componentDidUpdate() {
+    // Apply the new color in jQuery colorpicker
```
Url to bootstrap colorpicker

nasdan commented on 01 React Concepts/03 LifeCycle/tsconfig.json:
```diff
@@ -0,0 +1,17 @@
+    "sourceMap": true,
+    "noLib": false,
+    "suppressImplicitAnyIndexErrors": true,
+    "types": ["node"]
```
remove

nasdan commented on 01 React Concepts/03 LifeCycle/webpack.config.js:
```diff
@@ -0,0 +1,92 @@
+    ]
+  },
+  plugins: [
+    new ExtractTextPlugin({
```
Pending to add vendor plugin

crsanti committed:
Updated changes from comments:
- Extracted button to select a common color in a <Button/> component.
- Extracted form in a <FormColor /> component.
- Added color into <App /> state.
- Remove jQuery typings.
- Added webpack stuff.

crsanti committed:
Added reset form call when submit.
